src/GovUk.Education.ExploreEducationStatistics.Publisher/Functions/PublishReleaseContentFunction.cs

```diff
@@ -83,7 +89,15 @@ public class PublishReleaseContentFunction
 
                 await _contentService.DeletePreviousVersionsDownloadFiles(message.ReleaseId);
                 await _contentService.DeletePreviousVersionsContent(message.ReleaseId);
-                await _notificationsService.NotifySubscribers(message.ReleaseId);
+
+                var release = _contentDbContext.Releases
+                    .Include(r => r.ReleaseStatuses)
+                    .Single(r => r.Id == message.ReleaseId);
+                if (release.NotifySubscribers)
```

Comment:
I'd be tempted to do move this down within NotificationService and avoid the extra logic in the function classes with the additional database dependency. This will keep the function cleaner, remaining as a series of service calls.

We only send notifications for subscribers at a Publication level.

NotificationService works out the distinct Publications for the release id's passed to it, so this will change to work out the distinct Publications for Releases where notifications have been requested.

One other benefit of moving it there will be that you can also unit test it more easily.

It will also work exactly the same for `PublishStagedReleaseContentFunction.cs` where there are multiple release id's so no need for two different variations of it.

Reply:
I've moved the logic into NotificationService as you suggest. Do I also want to create unit tests for NotificationsService now?

Also I am tempted to rename NotificationsService#NotifySubscribers to something like `NotifySubscribersIfAppropriate` just so its clear this method isn't a guarantee that a notification will be sent.

Reply:
Yes I think now is probably a good time to add the unit tests in. I thought the same about the naming when writing this comment. Perhaps `NotifySubscribersIfApplicable`?